omod/src/main/java/org/openmrs/module/bedmanagement/rest/resource/BedVisitResource.java

```diff
@@ -0,0 +1,103 @@
+            bedDetails = bedManagementService.getBedDetailsByUuid(id);
+        return bedDetails;
+    }
+
+    @Override
+    protected void delete(BedDetails bedDetails, String reason, RequestContext requestContext) throws ResponseException {
+        throw new ResourceDoesNotSupportOperationException("delete of bed not supported in a visit");
+    }
+
+    @Override
+    public BedDetails newDelegate() {
+        return new BedDetails();
+    }
+
+    @Override
+    public BedDetails save(BedDetails bedDetails) {
+        throw new ResourceDoesNotSupportOperationException("save of bed not supported in a visit");
+    }
+
+    @Override
+    public void purge(BedDetails bedDetails, RequestContext requestContext) throws ResponseException {
+        throw new ResourceDoesNotSupportOperationException("purge of bed not supported in a visit");
+    }
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        if ((rep instanceof DefaultRepresentation) || (rep instanceof RefRepresentation)) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("bedId", "bedId");
+            description.addProperty("bedNumber", "bedNumber");
+            description.addProperty("bedType");
+            description.addProperty("physicalLocation", Representation.DEFAULT);
+            description.addProperty("patients", Representation.DEFAULT);
+            return description;
+        }
+        if ((rep instanceof FullRepresentation)) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("bedId", Representation.FULL);
+            description.addProperty("bedNumber", Representation.FULL);
+            description.addProperty("bedType");
+            description.addProperty("physicalLocation", Representation.FULL);
+            description.addProperty("patients", Representation.FULL);
+            return description;
+        }
+        return null;
+    }
+
+    @Override
+    public Object update(String uuid, SimpleObject propertiesToUpdate, RequestContext context) throws ResponseException {
+        throw new ResourceDoesNotSupportOperationException("update of bed not supported in a visit");
+    }
+
+    @Override
+    protected PageableResult doSearch(RequestContext context) {
+        BedManagementService bedManagementService = (BedManagementService) Context.getModuleOpenmrsServices(BedManagementService.class.getName()).get(0);
+        String visitUuid = context.getRequest().getParameter("visitUuid");
+        BedDetails bedDetails = bedManagementService.getLatestBedDetailsByVisit(visitUuid);
+        AlreadyPaged<BedDetails> alreadyPaged = new AlreadyPaged<BedDetails>(context, Arrays.asList(bedDetails), false);
+        return bedDetails == null || bedDetails.getBedId() == 0 ? super.doSearch(context) : alreadyPaged;
+    }
+}
```

Comment:
You are implementing a search handler for an existing resource. This is best done by just implementing the SearchHandler class.